### DISCUSSION TITLE
How to set nominal attributes for VDFT

### DISCUSSION THREAD

**Discussion by vinodkraman** [online-ml/river#913] [General]
Should I pass in a list nom_att with the indices corresponding to the columns of my dataset that are nominal attributes?

**Comment by smastelini** [comment]
Hi @vinodkraman, non-numerical features are automatically treated as categorical. The `nominal_attributes` parameter has to be set when the values of the nominal features are numeric but should be treated as nominal.

**Reply by smastelini** [reply]
If your categorical features are numeric (e.g., integers), you should pass the feature (column) names to `nominal_attributes` rather than their indices.

**Reply by vinodkraman** [reply]
Right, however what if I don't have a column names? Concretely, consider the input x =  {0:4, 1:2, 2:4, 3:2} where attributes 0, 1, 2, and 3 are are categorical. Would I need to specify nominal_attributes  = [0, 1, 2, 3] so that they match the keys of dictionary?

**Reply by smastelini** [reply]
Yes, exactly.

**Reply by vinodkraman** [reply]
Awesome, thanks!
